packages/python/accern_xyme/accern_xyme.py

```diff
@@ -891,18 +901,68 @@ def get_kafka_error_topic(self) -> str:
         assert res is not None
         return res
 
+    def get_kafka_error_message_topic(self) -> str:
+        res = cast(KafkaTopicNames, self.request_json(
+            METHOD_GET, "/kafka_topic_names", {}))["error_msg"]
+        assert res is not None
+        return res
+
     def delete_kafka_error_topic(self) -> KafkaTopics:
         return cast(KafkaTopics, self.request_json(
             METHOD_POST, "/kafka_topics", {
                 "num_partitions": 0,
             }))
 
-    def read_kafka_errors(self, offset: str = "current") -> List[str]:
+    def read_kafka_errors(
+            self,
+            consumer_type: ConsumerType,
+            offset: str = "current") -> List[str]:
+        if consumer_type == CONSUMER_DAG:
+            raise ValueError(
+                f"consumer_type cannot be {CONSUMER_DAG} "
+                "for reading kafka errors. provide consumer type from "
+                f"{[CONSUMER_ERR, CONSUMER_ERR_MSG]}")
         return cast(List[str], self.request_json(
             METHOD_GET, "/kafka_msg", {
                 "offset": offset,
+                "consumer_type": consumer_type
             }))
 
+    def read_kafka_full_json_errors(
+            self,
+            input_id_path: List[str]) -> Iterable[Tuple[str, Any]]:
+        errs = self.read_kafka_errors(consumer_type=CONSUMER_ERR)
+        msgs = self.read_kafka_errors(consumer_type=CONSUMER_ERR_MSG)
+
+        def parse_input_id_json(json_str: str) -> Optional[str]:
+            try:
+                res = json.loads(json_str)
+                for path in input_id_path:
+                    res = res[path]
+                return res
+            # except json.decoder.JSONDecodeError:
+            #     return None
+            except KeyError:
+                return None
```

Comment:
is it just KeyError or do we check for the possibility of json decode error as well and return None?
or Exception in general and return None?

Reply:
@JosuaKrause Can you check this

Reply:
yes we ignore json decode errors -- if there is non-json data we skip